pkg/network/dhcp/configurator.go

```diff
@@ -46,6 +46,10 @@ func NewConfigurator(cacheFactory cache.InterfaceCacheFactory, launcherPID strin
 	}
 }
 
+func (d *Configurator) SetDHCPStartedDIrectory(dhcpStartedDirectory string) {
+	d.dhcpStartedDirectory = dhcpStartedDirectory
+}
+
```

Comment:
Wouldn't it be simpler to just export the `d.dhcpStartedDirectory` ? Or even have a dedicated constructor for this. 

A setter will be an operation used just for this scenario, and will require 2 operations (create then mutate) / the exported variable will require 2 operations (create then mutate) .

My preference is a new constructor, the option I like least the one you have now.

Reply:
The idea was to modify the already created Configurator from https://github.com/kubevirt/kubevirt/blob/main/pkg/virt-launcher/virtwrap/network/podinterface.go#L67-L79 so we can exercise that part of the code and just modify what we need for tests, adding a new constructor does not fit well with that.

Reply:
What's the advantage of modifying the already created configurator ?... I fail to see the advantage of creating then mutating over creating it correctly.

Other then that... Why would you touch the production code ? It works. All is fine (in production). There's no requirement at all to configure where the dhcp started file goes.

Reply:
I agree with you I don't like either to add a mutator to configurator, but if we add a new constructor we have two options to use it at the virt-launcher/virtwrap/network tests:

- After calling NewPodNIC we overwrite dhcpConfigurator with the instance of the new constructor, but for that we have to be sure that we are also doing what is doing here https://github.com/kubevirt/kubevirt/blob/main/pkg/virt-launcher/virtwrap/network/podinterface.go#L67-L79
- Alternativily we enter at the parameterized factory function territory.

Reply:
Fair enough, let's meet in the middle then.